internal/data/inmemory/data.go

```diff
@@ -1,28 +1,30 @@
 package inmemory
 
 import (
-	"context"
-
+	"github.com/Red-Sock/Red-Cart/internal/data/inmemory/carts"
+	"github.com/Red-Sock/Red-Cart/internal/data/inmemory/users"
 	"github.com/Red-Sock/Red-Cart/internal/domain/user"
+	"github.com/Red-Sock/Red-Cart/internal/interfaces/data"
 )
 
-type Users struct {
+type Carts struct {
 	m map[int64]user.User
 }
 
-func New() *Users {
-	return &Users{
-		m: make(map[int64]user.User),
-	}
+type Storage struct {
+	Users *users.Users
+	Carts *carts.Carts
 }
 
-func (u *Users) Upsert(ctx context.Context, user user.User) error {
-
-	u.m[user.Id] = user
-	return nil
+func New() *Storage {
+	return &Storage{Users: users.NewUsers(),
```

Comment:
го по строчкам разнесём
```suggestion
	return &Storage{
	Users: users.NewUsers(),
```